### PR TITLE
feat(post): add upstreaming matters

### PR DESCRIPTION
**Pull request by RossComputerGuy**

### CODE REVIEW

File: content/post/why-upstreaming-matters/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+With packaging software, there will always be cases where software fail to run
+or build as expected. This is especially common in Nix where it is not a traditional
+Linux/UNIX environment. For example, Nix does not use the filesystem hierarchy (FHS)
+which is standard in POSIX. Instead each package has its own directory in the Nix
+store, also referred to as a local cache, and contains directories such as `bin`,
+`lib`, `share`, etc. As most software is written to expect the FHS, they can easily
+break and not run as expected. The Nix store is also writable at build time of a
+package but ready only after that. This means the Nix store is essentially immutable.
```
Comment (Enstrayed) [inline content/post/why-upstreaming-matters/index.md]: Alternative phrasing: `The Nix store is only writable at build time, becoming essentially immutable thereafter.`

File: content/post/why-upstreaming-matters/index.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+These factors cause many software ranging from LLVM to Google Chrome and Flutter to
+either not build correctly or not work as intended.
+
+When packaging software and you have to make a patch, it is best to send these patches
+upsteam to the official source code repository of a package. This offers a collaborative
+effort to ensure the software builds in one environment but also other environments as well.
```
Comment (Enstrayed) [inline content/post/why-upstreaming-matters/index.md:24]: This paragraph reads somewhat weird. Alternative phrasing: `When a patch is required during packaging, it's a good idea to send that patch to the upstream repository. This helps in both preventing the necessity of the patch when packaging future versions of the software, and notifying the upstream contributors that special actions must be taken for a given distro.`

File: content/post/why-upstreaming-matters/index.md (hunk 1 above)
Comment (RossComputerGuy) [reply content/post/why-upstreaming-matters/index.md]: Alright, rephrased it.

File: content/post/why-upstreaming-matters/index.md (hunk 2 above)
Comment (RossComputerGuy) [reply content/post/why-upstreaming-matters/index.md:24]: Rephrased.

**Review by Enstrayed — APPROVED** [review]

**Pull request merged**